org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintProjectEventListener.java

```diff
@@ -33,22 +39,39 @@
   @Override
   public void resourceChanged(IResourceChangeEvent event) {
     if (event.getType() == IResourceChangeEvent.POST_CHANGE) {
+      List<ISonarLintProject> projectToSubscribeToNotifications = new ArrayList<>();
+      List<ISonarLintProject> projectToUnsubscribeFromNotifications = new ArrayList<>();
       try {
-        event.getDelta().accept(SonarLintProjectEventListener::visitDelta);
+        event.getDelta().accept(delta -> visitDelta(delta, projectToSubscribeToNotifications, projectToUnsubscribeFromNotifications));
       } catch (CoreException e) {
         SonarLintLogger.get().error(e.getMessage(), e);
       }
+      if (!projectToSubscribeToNotifications.isEmpty() || !projectToUnsubscribeFromNotifications.isEmpty()) {
+        Job subscribeToNotificationsJob = new Job("Subscribe to notifications") {
+          @Override
+          protected IStatus run(IProgressMonitor monitor) {
+            if (!projectToSubscribeToNotifications.isEmpty()) {
+              SonarLintCorePlugin.getInstance().notificationsManager().subscribeToNotifications(projectToSubscribeToNotifications,
```

Comment:
Is it possible that we subscribe twice, e.g. if a project was already subscribed and was edited ?

Reply:
Normally, thanks to `delta.getFlags() & IResourceDelta.OPEN`, we are only notified about project open/close events.